tests/issues/1008/record.smt2

```diff
@@ -0,0 +1,8 @@
+(set-logic ALL)
+(declare-datatype t ((B (i Int))))
+
+(declare-const e t)
+
+(assert ((_ is B) e))
+(assert (forall ((n Int)) (distinct e (B n))))
```

Comment:
I think we should add a note somewhere that "this currently returns `unknown` but we want it to return `unsat` once #1008 is fixed", to avoid thinking there is a soundness issue with the fix.

Reply:
I will add the note in #1008.

Reply:
The rationale for adding a note in the test itself is that if a test fails, the test file is the first place we are going to look.